src/basic/fileio.c

```diff
@@ -1187,3 +1187,25 @@ int warn_file_is_world_accessible(const char *filename, struct stat *st, const c
                             filename, st->st_mode & 07777);
         return 0;
 }
+
+int sync_rights(FILE *from, const char *to) {
```

Comment:
Hmm, FILE* as input in a generic function is a bit weird... maybe take an "int fd" here, and use the fileno() on the caller's side?

i.e. stuff that makes sense in a local, static helper function might be a bit too specific to keep 1:1 when copied into the generic src/basic/ tree

Reply:
Made both arguments file descriptors and modified `sync_rights` to use `fchmod_and_chown`.